### ISSUE TITLE
test: Add RenderToMarkup and RenderObjectToMarkup

### ISSUE THREAD

**Issue by flimzy** [fyne-io/fyne#5124]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same feature, before opening a new one.
- [X] This issue only relates to a single feature. I will open new issues for any other features.

### Is your feature request related to a problem?

I would like versions of [AssertRendersToMarkup](https://github.com/fyne-io/fyne/blob/v2.5.1/test/test.go#L104) and [AssertObjectRendersToMarkup](https://pkg.go.dev/fyne.io/fyne/v2@v2.5.1/test#AssertObjectRendersToMarkup) that don't do any assertions, or use golden files. I'd like them just to return the markup version of the object/canvas, which I can then do my own assertions on.

This would give me the flexibility to use regular expressions, or other checks that don't consider the entire output.

### Is it possible to construct a solution with the existing API?

No.

### Describe the solution you'd like to see.

  I propose the following additional functions be added to `fyne.io/fyne/v2/test`

```golang
RenderToMarkup(c fyne.Canvas) string
RenderObjectToMarkup(o fyne.CanvasObject) string
```

**Comment by andydotxyz** [comment]
Seems like a good request

**Comment by andydotxyz** [comment]
Thanks for the addition @flimzy so this issue is done.
It is landed on develop and will be in v2.6.0 which should be around the close of the year.
